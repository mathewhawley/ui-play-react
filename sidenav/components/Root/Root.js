import React, { Component } from 'react';
import Header from '../Header';
import Nav from '../Nav';

class Root extends Component {
  constructor() {
    super();
    this.state = {
      active: false,
    };
    this.toggleNav = this.toggleNav.bind(this);
  }

  toggleNav() {
    this.setState({ active: !this.state.active });
  }

  render() {
    return (
      <div>
        <Header toggleNav={this.toggleNav} />
        <Nav {...this.state} toggle={this.toggleNav} />
      </div>
    );
  }
}

export default Root;
