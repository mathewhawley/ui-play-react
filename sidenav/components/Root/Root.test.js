/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import renderer from 'react-test-renderer';
import Root from './Root';

const mockState = (overrides) => {
  return {
    active: false,
    transitioning: false,
    isDragging: false,
    startX: 0,
    currentX: 0,
    translateX: 0,
    ...overrides,
  };
};

describe('<Root />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Root />);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Root />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should stop click event propagation when panel is clicked', () => {
    const panel = wrapper.find('.panel');
    const spy = sinon.spy();
    panel.simulate('click', { stopPropagation: spy });
    expect(spy.calledOnce).toBe(true);
  });

  it('should correctly update the state on `showSideNav()`', () => {
    const stateAfter = mockState({
      active: true,
      transitioning: true,
      isDragging: false,
    });
    wrapper.instance().showSideNav();
    expect(wrapper.state()).toEqual(stateAfter);
  });

  it('should correctly update the state on `hideSideNav()`', () => {
    const stateAfter = mockState({
      active: false,
      transitioning: true,
      isDragging: false,
    });
    wrapper.instance().hideSideNav();
    expect(wrapper.state()).toEqual(stateAfter);
  });

  it('should correctly update the state on `onTransitionEnd()`', () => {
    const stateAfter = mockState({
      active: true,
    });
    wrapper.instance().showSideNav();
    wrapper.instance().onTransitionEnd();
    expect(wrapper.state()).toEqual(stateAfter);
  });
});
