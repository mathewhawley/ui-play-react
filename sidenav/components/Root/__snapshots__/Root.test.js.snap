exports[`<Root /> renders correctly 1`] = `
<div>
  <div
    className="base">
    <button
      className="base button material-icons"
      onClick={[Function]}>
      menu
    </button>
  </div>
  <div
    className="base"
    onClick={[Function]}>
    <nav
      className="panel">
      <button
        className="base material-icons"
        onClick={[Function]}>
        close
      </button>
    </nav>
  </div>
</div>
`;
